From: Joel Selvaraj <jo@jsfamily.in>
Date: Mon, 6 Dec 2021 12:46:14 +0530
Subject: arm64: dts: qcom: sdm845-xiaomi-beryllium-ebbg: introduce support
 for display and touchscreen

The EBBG variant uses Focaltech touchscreen and ebbg panel.
---
 .../boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts | 145 +++++++++++++++++++++
 1 file changed, 145 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts
index 6d3932a..3d31e95 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-ebbg.dts
@@ -3,3 +3,148 @@
 /dts-v1/;
 
 #include "sdm845-xiaomi-beryllium-common.dtsi"
+
+/ {
+	/*
+	 * I don't know where the heck this regulator comes from
+	 * but it powers the touchscreen vdd rail (vsn/vsp??)
+	 */
+	ts_vdd: ts-vdd-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "ts_vdd";
+		regulator-min-microvolt = <3000000>;
+		regulator-max-microvolt = <3000000>;
+	};
+};
+
+&tlmm {
+
+	sde_dsi_active: sde_dsi_active {
+		mux {
+			pins = "gpio6", "gpio52";
+			function = "gpio";
+		};
+
+		config {
+			pins = "gpio6", "gpio52";
+			drive-strength = <8>;   /* 8 mA */
+			bias-disable = <0>;   /* no pull */
+		};
+	};
+
+	sde_dsi_suspend: sde_dsi_suspend {
+		mux {
+			pins = "gpio6", "gpio52";
+			function = "gpio";
+		};
+
+		config {
+			pins = "gpio6", "gpio52";
+			drive-strength = <2>;   /* 2 mA */
+			bias-pull-down;		/* PULL DOWN */
+		};
+	};
+};
+
+
+&display_panel {
+
+	status = "okay";
+
+	reset-gpios = <&tlmm 6 GPIO_ACTIVE_HIGH>;
+
+	lab-supply = <&lab>;
+	ibb-supply = <&ibb>;
+
+	pinctrl-names = "panel_active", "panel_suspend";
+	pinctrl-0 = <&sde_dsi_active>;
+	pinctrl-1 = <&sde_dsi_suspend>;
+
+	compatible = "ebbg,ft8719";
+
+};
+
+&tlmm {
+	ts_mux {
+
+		ts_int_active: ts_int_active {
+			mux {
+				pins = "gpio31";
+				function = "gpio";
+			};
+
+			config {
+				pins = "gpio31";
+				drive-strength = <16>;
+				bias-pull-down;
+				input-enable;
+			};
+		};
+
+		ts_reset_active: ts_reset_active {
+			mux {
+				pins = "gpio32";
+				function = "gpio";
+			};
+
+			config {
+				pins = "gpio32";
+				drive-strength = <16>;
+				output-high;
+			};
+		};
+
+		ts_reset_suspend: ts_reset_suspend {
+			mux {
+				pins = "gpio32";
+				function = "gpio";
+			};
+
+			config {
+				pins = "gpio32";
+				drive-strength = <2>;
+				bias-disable;
+				output-low;
+			};
+		};
+
+		ts_int_suspend: ts_int_suspend {
+			mux {
+				pins = "gpio31";
+				function = "gpio";
+			};
+
+			config {
+				pins = "gpio31";
+				drive-strength = <2>;
+				bias-pull-down;
+				input-enable;
+			};
+		};
+	};
+};
+
+&i2c14 {
+	status = "okay";
+
+	touchscreen@38 {
+		compatible = "focaltech,fts8719";
+		reg = <0x38>;
+		wakeup-source;
+		interrupt-parent = <&tlmm>;
+		interrupts = <31 0>;
+		vdd-supply = <&ts_vdd>;
+		vcc-i2c-supply = <&vreg_l14a_1p8>;
+
+		pinctrl-names = "default", "suspend";
+		pinctrl-0 = <&ts_int_active &ts_reset_active>;
+		pinctrl-1 = <&ts_int_suspend &ts_reset_suspend>;
+
+		reset-gpio = <&tlmm 32 0x01>;
+		irq-gpio = <&tlmm 31 0x02>;
+		touchscreen-size-x = <1080>;
+		touchscreen-size-y = <2246>;
+		focaltech,max-touch-number = <10>;
+	};
+};
+
